client.go

```diff
@@ -1006,7 +1016,11 @@ func (f *File) Read(b []byte) (int, error) {
 	f.mu.Lock()
 	defer f.mu.Unlock()
 
-	n, err := f.ReadAt(b, f.offset)
+	if f.handle == "" {
+		return 0, os.ErrClosed
+	}
+
+	n, err := f.readAt(b, f.offset)
```

Comment:
This function has to change names, so that `ReadAt` can read-lock, while this function needs to hold the write-lock.

So, `readat` needs to be called with a lock, but it doesn’t matter which lock.